client/src/components/new-request/quick-search-results.vue

```diff
@@ -0,0 +1,135 @@
+<template>
+  <MainContainer id="quick-search-results-container">
+    <template v-slot:container-header>
+      <v-col cols="auto" class="h6 py-0 mt-1">
+        You are searching for a name for a
+        {{ entityText === ' BC Corporation' && location.text === ' BC' ? '' : ' ' + location.text }}
+        {{ entityText }}
+      </v-col>
+    </template>
+    <template v-slot:content>
+      <v-row no-gutters class="pt-3">
+        <NameInput/>
+      </v-row>
+      <v-row justify="center" class="ma-0 pa-0">
+        <v-col cols="12" class="quick-search-title">
+          <b>Quick Search Results</b>
+        </v-col>
+      </v-row>
+      <v-row v-if="namesList.length" justify="center" class="ma-0 pa-0">
+        <v-col cols="12" class="quick-search-info">
+          We found <b>{{namesList.length}}</b> names using these, or similar, words.
+        </v-col>
+        <v-col cols="12" class="quick-search-info">
+          Names cannot be too similar to existing names within the same business category.
+          Try and request a unique name.
+        </v-col>
+        <v-col cols="12" class="names-list">
+          <QuickSearchNames/>
+        </v-col>
+      </v-row>
+      <v-row v-else justify="center">
+        <v-col cols="12" class="quick-search-info">
+          We did not find any names using these exact or similar words within the same business category.
+        </v-col>
+      </v-row>
+      <v-row justify="center">
+        <v-col cols="auto" class="pb-0 pr-1">
+          <v-btn id="search-again-button" :disabled="nameChanged" @click="searchAgain">Search Again</v-btn>
+        </v-col>
+        <v-col cols="auto" class="pb-0 pr-1">
+          <v-btn class="continue-search-button" color="primary" outlined @click="detailedSearch">
+            Continue to Detailed Analysis
+          </v-btn>
+        </v-col>
+      </v-row>
+    </template>
+  </MainContainer>
+</template>
+
+<script lang="ts">
+import MainContainer from '@/components/new-request/main-container.vue'
+import NameInput from '@/components/new-request/name-input.vue'
+import QuickSearchNames from '@/components/new-request/quick-search-names.vue'
+import allDesignations, { allDesignationsList } from '@/store/list-data/designations'
+import newReqModule from '@/store/new-request-module'
+import { Component, Vue, Watch } from 'vue-property-decorator'
+
+@Component({
+  components: { MainContainer, NameInput, QuickSearchNames }
+})
+export default class QuickSearchResults extends Vue {
+  config = {
+    modules: {
+      toolbar: false
+    },
+    placeholder: '',
+    scrollingContainer: false
+  }
+  contents: string = ''
+  originalName: string = ''
+
+  get entity_type_cd () {
+    return newReqModule.entity_type_cd
+  }
+  get entityText () {
+    return newReqModule.entityTextFromValue
+  }
+  get location () {
+    let value = newReqModule.location
+    let options = newReqModule.locationOptions
+    return options.find((opt: any) => opt.value === value)
```

Comment:
Is it OK if this returns undefined or null?

Reply:
I just checked. If the value is not found then the HTML will throw an error. Either return a fallback object here or handle it in the HTML please.

Reply:
hmm okay. This is the same code that we have in the other components, but I will see how they are handfling it

Reply:
hey sev, this is an existing error in the analyze components used in the top section of each one so to fix it properly it should be in its own comp and fixed there, then imported into each one. Can we say tech debt ticket for this?

Reply:
Yeah let's do that for all components that can fail this way.